src/main/java/com/redhat/devtools/intellij/qute/psi/utils/PsiQuteProjectUtils.java

```diff
@@ -104,7 +103,7 @@ private static String getTemplateBaseDir(Module javaProject) {
     public static @Nullable VirtualFile findBestResourcesDir(@NotNull Module javaProject) {
         List<VirtualFile> resourcesDirs = ModuleRootManager.getInstance(javaProject).getSourceRoots(JavaResourceRootType.RESOURCE);
         if (!resourcesDirs.isEmpty()) {
-            Collections.sort(resourcesDirs, ROOT_COMPARATOR); // put root with smallest path first (eliminates generated sources roots)
+            QuarkusModuleUtil.sortRoot(resourcesDirs); // put root with smallest path first (eliminates generated sources roots)
             // The module configure 'Resources folder'
             // 1) loop for each configured resources dir and returns the first which contains 'templates' folder.
             for (var dir : resourcesDirs) {
```

Comment:
L118 to L122 can be removed, it's already covered by QuarkusModuleUtil.getModuleDirPath

Reply:
Good cach! I wanted to do that, but I forget it while my refactoring :)

fixed.